src/containers/Form.js

```diff
@@ -1,36 +1,82 @@
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import * as Actions from '../actions';
-import { InputField } from '../components';
+import { InputField, DaySelect, TypeSelect } from '../components';
+import { withState, compose, range } from '../utils';
+import {
+  isValidText,
+  isValidImageUrl,
+  isValidSpotifyUrl,
+  isValidYouTubeUrl
+} from '../utils/validation';
+import { windowTypes } from '../constants';
+
+function validate({ text, type }) {
+  switch (type) {
+  case windowTypes.TEXT:
+    return isValidText(text) ? null : 'Invalid text';
+  case windowTypes.IMAGE:
+    return isValidImageUrl(text) ? null : 'Invalid image url';
+  case windowTypes.SPOTIFY:
+    return isValidSpotifyUrl(text) ? null : 'Invalid Spotify url';
+  case windowTypes.YOUTUBE:
+    return isValidYouTubeUrl(text) ? null : 'Invalid YouTube url';
+  default:
+    return 'Invalid type';
+  }
+}
+
+const initialState = { text: '', type: 'text' };
+
+function Form({ validDays, createWindow, state, updateState }) {
+  const handleTypeChange = type => {
```

Comment:
Jeg syns ikke det er noe mer magisk enn at vi får f.eks. `createWindow` fra `connect`. Det vil uansett handle om å kjenne API-et man bruker - `withState` i dette tilfellet - men jeg er såklart åpen for forbedringer :)